File: leetcode_ly/java/Hash Table/FindTheDifference.java
public class Solution {
    public char findTheDifference(String s, String t) {
        int[] res = new int[26];
        for(int i = 0; i < s.length(); i++){
            res[s.charAt(i) - 'a']++;
        }
        int j;
        for(j = 0; j < t.length(); j++){
            if(res[t.charAt(j) - 'a'] > 0) res[t.charAt(j) - 'a']--;
            else return t.charAt(j);
        }
        return t.charAt(j);
    }
}

File: leetcode_ly/c++/HashTable/FindTheDifference.cpp
class Solution {
public:
    char findTheDifference(string s, string t) {
        map<int,int> myMap;
        for(int i = 0; i < s.size(); i++){
            myMap[s[i] - 'a']++;
        }
        for(int i = 0; i < t.size(); i++){
        	myMap[t[i] - 'a']--;
        }
        map<int,int>::iterator it;
        for(it = myMap.begin(); it != myMap.end(); it++){
            if(it->second) return 'a' + it->first;
        }
        return t[0];
    }
};